packages/services/schema/src/lib/federation-tag-extraction.spec.ts

```diff
@@ -126,7 +68,12 @@ describe('applyTagFilterToInaccessibleTransformOnSubgraphSchema', () => {
       `);
 
       const outputSdl = print(
-        applyTagFilterToInaccessibleTransformOnSubgraphSchema(sdl, filter).typeDefs,
+        applyTagFilterToInaccessibleTransformOnSubgraphSchema(
+          sdl,
+          filter,
+          'federation__inaccessible',
+          'federation__tag',
+        ).typeDefs,
```

Comment:
The need to pass in these directive names vs extract then _again_ from the typedefs accounts for `205` additional lines, which makes it look like the utility didnt help simplify much.... but after taking this fact into account, we realize this change saved approx 220 lines of meaningful code, and makes it possible to support multiple versions of implementations should it ever be necessary.

Reply:
I moved the link extraction into this federation tag extraction function because for those typedefs it wasnt being used elsewhere and it made this diff much nicer.